src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,238 @@
+                y_vel_target = ((self._roomba_point.point.y + y_overshoot)
+                                    * self._K_Y + roomba_y_velocity)
+
+                z_vel_target = self._z_holder.get_height_hold_response(
+                    self._drone_odometry.pose.pose.position.z)
+
+                #caps velocity
+                vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+                if vel_target > self._MAX_HORIZ_SPEED:
+                    x_vel_target = x_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                    y_vel_target = y_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target =  math.copysign(self._MAX_Z_VELOCITY, z_vel_target)
+
+                desired_vel = [x_vel_target, y_vel_target, z_vel_target]
+
+                drone_vel_x = self._drone_odometry.twist.twist.linear.x
+                drone_vel_y = self._drone_odometry.twist.twist.linear.y
+                drone_vel_z = self._drone_odometry.twist.twist.linear.z
+
+                if self._current_velocity is None:
+                    self._current_velocity = [drone_vel_x, drone_vel_y, drone_vel_z]
+
+                desired_vel = self._limiter.limit_acceleration(self._current_velocity, desired_vel)
+
+                velocity = TwistStamped()
+                velocity.header.frame_id = 'level_quad'
+                velocity.header.stamp = rospy.Time.now()
+                velocity.twist.linear.x = desired_vel[0]
+                velocity.twist.linear.y = desired_vel[1]
+                velocity.twist.linear.z = desired_vel[2]
+
+                self._current_velocity = desired_vel
+
+                return (TaskRunning(), VelocityCommand(velocity))
+
+            return (TaskAborted(msg='Illegal state reached in GoTo Roomba task'),)
+
+    ## checks to see if passed in roomba id is in sight of quad
+    def _check_roomba_in_sight(self):
+        for odometry in self._roomba_array.data:
+            if odometry.child_frame_id == self._roomba_id:
+                self._roomba_odometry = odometry
+                return True
+        return False
+
+    # that the drone and roomba are both within a specified distance
+    # in order to start/continue the task
+    def _check_max_roomba_range(self):
+        _distance_to_roomba = math.sqrt(self._roomba_point.point.x**2 +
+                            self._roomba_point.point.y**2)
+
+        return (_distance_to_roomba <= self._MAX_START_TASK_DIST + self._overshoot)
+
+    # checks the radial distance from the drone to the roomba
+    # in order to end the task as successful
+    def _check_max_ending_roomba_range(self):
+        _distance_to_roomba = math.sqrt(self._roomba_point.point.x**2 +
+                            self._roomba_point.point.y**2)
+
+        return (_distance_to_roomba <= (self._ending_radius_to_roomba + self._overshoot))
```

Comment:
You named the ending distance variable differently above. Make sure they are the same otherwise you are comparing your distance to who knows what.